src/pages/workflows/workflow/common/WorkflowModal.tsx

```diff
@@ -0,0 +1,110 @@
+import { ButtonPrimary, Modal } from '@terra-ui-packages/components';
+import React from 'react';
+import { TextInput } from 'src/components/input';
+import { FormLabel } from 'src/libs/forms';
+
+interface WorkflowModalProps {
+  setCreateWorkflowModalOpen: (x: boolean) => void;
+  title: string;
+  namespace: string;
+  name: string;
+  synopsis: string;
+  buttonAction: string; // name of the primary button i.e. 'save' or 'upload'
+  setWorkflowNamespace: (value: string) => void;
+  setWorkflowName: (value: string) => void;
+  setWorkflowSynopsis: (value: string) => void;
+}
+
+interface NamespaceNameSectionProps {
+  namespace: string | undefined;
+  name: string | undefined;
+  setWorkflowNamespace: (value: string) => void;
+  setWorkflowName: (value: string) => void;
+}
+
+interface SynopsisSnapshotSectionProps {
+  synopsis: string;
+  setWorkflowSynopsis: (value: string) => void;
+}
+
+const NamespaceNameSection = (props: NamespaceNameSectionProps) => {
+  const namingMessage = 'Only letters, numbers, underscores, dashes, and periods allowed';
+  const { namespace, name, setWorkflowNamespace, setWorkflowName } = props;
+
+  return (
+    <>
```

Comment:
Out of curiosity, what does this empty braces do here? 🤔

Reply:
From my understanding, it acts like `Fragment`. Since I'm not returning a single `div` element, I have to use the braces for the component to render.